lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -743,6 +758,54 @@ modifyLedgerBody f (Cardano.Tx body keyWits) = Cardano.Tx body' keyWits
                         auxData
                         validity
 
+
+emptyTx
+    :: RecentEra era
+    -> Core.Tx (ShelleyLedgerEra era)
+emptyTx RecentEraBabbage = Alonzo.ValidatedTx (emptyTxBody RecentEraBabbage)
+    mempty
+    (Babbage.IsValid True)
+    SNothing
+emptyTx RecentEraAlonzo = Alonzo.ValidatedTx (emptyTxBody RecentEraAlonzo)
+    mempty
+    (Babbage.IsValid True)
+    SNothing
+
+emptyTxBody
+    :: RecentEra era
+    -> Core.TxBody (ShelleyLedgerEra era)
+emptyTxBody RecentEraBabbage = Babbage.TxBody
+    mempty
+    mempty
+    mempty
+    mempty
+    SNothing
+    SNothing -- total collateral
+    mempty
+    (Wdrl mempty)
+    (Coin 0) -- fee
+    (ValidityInterval SNothing SNothing)
+    SNothing
+    mempty
+    mempty
+    SNothing
+    SNothing
+    SNothing
```

Comment:
I'm surprised that `cardano-ledger` doesn't define an `emptyTxBody` or equivalent. Perhaps we could make a PR? (Though it'd probably be ages before we could use it. 😄)

Reply:
I think there will be functions we can use once we bump the ledger version, probably like this: https://github.com/input-output-hk/cardano-ledger/blob/9af95e8812224f42f6fe6fdbb8d5b2e8c85e6260/eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs#L487-L502